test/module/irohad/consensus/yac/mock_yac_network.hpp

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_MOCK_YAC_NETWORK_HPP
+#define IROHA_MOCK_YAC_NETWORK_HPP
+
+#include <gmock/gmock.h>
+
+#include "consensus/yac/transport/yac_network_interface.hpp"
+
+namespace iroha {
+  namespace consensus {
+    namespace yac {
+      class MockYacNetwork : public YacNetwork {
+       public:
+        void subscribe(
+            std::shared_ptr<YacNetworkNotifications> handler) override {
+          notification = handler;
+        };
+
+        void release() {
+          notification.reset();
+        }
+
+        MOCK_METHOD2(sendState,
+                     void(const shared_model::interface::Peer &,
+                          const std::vector<VoteMessage> &));
+
+        MockYacNetwork() = default;
+
+        MockYacNetwork(const MockYacNetwork &rhs)
+            : notification(rhs.notification) {}
+
+        MockYacNetwork &operator=(const MockYacNetwork &rhs) {
+          notification = rhs.notification;
+          return *this;
+        }
+
+        MockYacNetwork(MockYacNetwork &&rhs) {
+          std::swap(notification, rhs.notification);
+        }
+
+        MockYacNetwork &operator=(MockYacNetwork &&rhs) {
+          std::swap(notification, rhs.notification);
+          return *this;
+        }
+
+        std::shared_ptr<YacNetworkNotifications> notification;
```

Comment:
Maybe declare it as private?

Reply:
I don't see a reason to do that. Could you explain why the change is required?

Reply:
This member is not used anywhere but in this class, so just for encapsulation purposes